snuba/settings/__init__.py

```diff
@@ -78,16 +83,29 @@
 CLICKHOUSE_TRACE_USER = os.environ.get("CLICKHOUSE_TRACE_USER", "default")
 CLICKHOUSE_TRACE_PASSWORD = os.environ.get("CLICKHOUSE_TRACE_PASS", "")
 
-# Redis Options
 USE_REDIS_CLUSTER = os.environ.get("USE_REDIS_CLUSTER", "0") != "0"
 
+# Redis Options
 REDIS_CLUSTER_STARTUP_NODES = None
 REDIS_HOST = os.environ.get("REDIS_HOST", "localhost")
 REDIS_PORT = int(os.environ.get("REDIS_PORT", 6379))
 REDIS_PASSWORD = os.environ.get("REDIS_PASSWORD")
 REDIS_DB = int(os.environ.get("REDIS_DB", 1))
 REDIS_INIT_MAX_RETRIES = 3
 
+# Set self-hosted users up with the default setup where everything is
+# on one redis. Users who want to multiplex to multiple clusters should
+# fill in the values of REDIS_CLUSTER_MAP with lists of NodeDescriptor[s]
+REDIS_SINGLE_NODE_DESCRIPTOR = NodeDescriptor(
+    REDIS_HOST, REDIS_PORT, REDIS_DB, REDIS_PASSWORD
+)
+REDIS_CLUSTER_MAP: Mapping[ClusterFunction, ConnectionDescriptor] = {
```

Comment:
I don't think you should put these enums and classes in here. I think the settings file should be simple JSON or something equivalent that the redis files pull in and build into the various connection classes. That solves the import cycle as well.

Reply:
OK that makes sense, maybe we could add this developer guideline and other settings stuff to https://getsentry.github.io/snuba/index.html

One thing I really wasn't sure about is how we might deprecate a setting (which IMO we want to be able to do)

Additionally things like "don't send datadog metrics in consumers without setting a sample rate" etc.